### PR TITLE
only show restrictions when they're required

### PR DESCRIPTION
**Pull request by jamesgorrie**
Inspiration:
* Talking to Jonny
* Talking to Joanna
* Looking through the data and finding lots of "No restrictions"

This only shows the restrictions box if they are required or you explicitly say that you want to add some.

restrictions-ui

### CODE REVIEW

**Comment by jamesgorrie** [comment]
As @theefer noticed, it shows and requires the restrictions on restricted categories.

File: kahuna/public/js/usage-rights/usage-rights-editor.html [hunk 1]
```diff
@@ -39,94 +38,113 @@
...
                         name="{{ property.name }}"
-                        ng:if="property.optionsMap"
                         ng:model="ctrl.model[property.name]"
-                        ng:required="property.required"
-                        ng:options="o for o in ctrl.getOptionsFor(property)">
-                        <option value="">None</option>
-                    </select>
-
+                        ng:switch-when="text"
+                        ng:required="property.required">
```
Comment (theefer) [inline kahuna/public/js/usage-rights/usage-rights-editor.html]: Stupid question: having this whitespace in the textarea doesn't affect its initial value, or does it?

**Comment by theefer** [comment]
:+1: 

Bonus: would be nice to focus the restrictions field when checking the "restricted" checkbox, but these things are often fiddly in Angular so not worth spending too much time on.

**Comment by jamesgorrie** [comment]
I want to see how useful this is first and then I'll work on the focus.

**Pull request merged**